docs/DeveloperGuide.md

```diff
@@ -144,92 +139,88 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
-
-![CommitActivityDiagram](images/CommitActivityDiagram.png)
-
-#### Design consideration:
-
-##### Aspect: How undo & redo executes
-
-* **Alternative 1 (current choice):** Saves the entire address book.
-  * Pros: Easy to implement.
-  * Cons: May have performance issues in terms of memory usage.
-
-* **Alternative 2:** Individual command knows how to undo/redo by
-  itself.
-  * Pros: Will use less memory (e.g. for `delete`, just save the person being deleted).
-  * Cons: We must ensure that the implementation of each individual command are correct.
-
-_{more aspects and alternatives to be added}_
-
-### \[Proposed\] Data archiving
-
-_{Explain here how the data archiving feature will be implemented}_
+### Contact and tag management
+![contact_tag_diagram](images/ContactTagDiagram.png)
+
+##### General Design
+**`Person`** component: 
+
+In Athena, contacts are represented by `Person` objects. `Person` objects have several properties such as email, address, etc. However, a `Person` can also be tagged with multiple `Tag`s.
+- `AddressBook` handles all direct matters concerning `Person` objects. It has a `TagManager` and `UniquePersonList`.  
+- `UniquePersonList` keeps track of all `Person` objects. Using `Person` class' `isSamePerson(Person)` method, it ensures that there are no duplicate contacts in Athena.
+- `TagManager` keeps track of which contacts contain which tags. It uses a hash map, with `Tag`s as keys and stores a set of `Person`s as its value.
+
+All manipulation of `Person` objects have to be done through `AddressBook`. `AddressBook` provides simple methods that can be used by higher-level components such as 
+- `void removePerson(Person)`
+- `void addPerson(Person)`
+- `boolean hasPerson(Person)`
+- and more...
+
+**`Tag`** component:
+
+`Tag`s are represented by a single alphanumeric string with no spaces. In v1.3, support for child-tagging was implemented. This allows directional relations to be established between `Tag`s. Certain commands also affect a `Tag` and its child-tags.
+- `TagTree` handles tag-to-tag relations 
+- `TagTreeImpl` extends from the abstract class `TagTree`. It uses a tree data structure to store directional tag-to-tag relations. The implementation of the tree is done with 
+a hash map, with `Tag`s used as keys, and its value being a set of all its child-tags.
+
+Any new links established between tags have to go through the `TagTree`. `TagTree` provides several simple methods such as 
+- `void addSubTagTo(Tag tag)`  
+- `boolean hasTag(Tag tag)`  
+- and more...
+
+**`Integration`** component:
+
+The `ContactTagIntegrationManager` class provides a few defined methods that affect both `Person`s and `Tag`s together. Methods include:
+- `void deleteTag(Tag)`
+- `void deleteTagRecursive(Tag)` - deletes a `Tag` and all its sub-tags
+- and more...
+
+The above methods listed methods are shown as examples to illustrate the difficulty of preserving consistency within the system.
```

Comment:
"The methods listed above illustrate the difficulty of preserving consistency within the system."